### ISSUE TITLE
Accept bytearray in place of bytes

### ISSUE THREAD

**Issue by alexforencich** [secdev/scapy#3015]
#### Brief description

Currently, scapy does not accept `bytearray`s for payload data, likely among other things:

    from scapy.all import *
    pkt = Ether() / bytearray(10)

results in

    AttributeError: 'bytearray' object has no attribute '__rdiv__'

Manually casting to `bytes` works fine, but IMO `bytes` and `bytearray` should be interchangeable, not to mention other objects can implement `__bytes__`, so perhaps revisiting the implementation of `Packet.__div__` to either accept `bytearray`s in addition to `bytes`, or to attempt to cast the argument to `bytes` and only call `__rdiv__` if that fails.  

#### Environment

-   Scapy version: `2.4.3`
-   Python version: `3.9.0`
-   Operating System: `Arch`

#### How to reproduce

    from scapy.all import *
    Ether() / bytearray(10)

#### Actual result

    AttributeError: 'bytearray' object has no attribute '__rdiv__'

#### Expected result

    <Ether  |<Raw  load='\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' |>>

**Comment by gpotter2** [comment]
Hi and thanks for this report.
What do you think of https://github.com/secdev/scapy/pull/3017 ? Thanks

**Comment by alexforencich** [comment]
That should work, but I'm just wondering if there is anything to be said for doing something like

    else:
        try:
            return self / conf.raw_layer(load=other)
        except TypeError:
            return other.__rdiv__(self)  # type: ignore

and as such "do the right thing" with objects that can be converted to bytes via `__bytes__`, etc.  Looks like `load` should get passed through `bytes_encode` in `StrField`, and that throws a `TypeError` if the argument cannot be converted to bytes.

Also, this may need to be addressed in a few other places.  It looks like `Packet.add_payload()` and `Packet.__rdiv__()` also check for bytes, and as such should a similar fix.

**Comment by gpotter2** [comment]
There likely won't be an error until build. I'd rather also control the types specifically, to make sure people aren't tempted to use it wrong.

**Comment by alexforencich** [comment]
Oh, so the actual conversion is deferred?  What about explicitly casting to bytes like so:

    else:
        try:
            return self / conf.raw_layer(load=bytes(other))
        except TypeError:
            pass
        return other.__rdiv__(self)  # type: ignore

I just tested that myself, seems to work.  I moved `return other.__rdiv__(self)` outside of the `try` block so python only prints out one exception.

But sure, I also understand if you want to be non-pythonic with type handling to be more consistent with how it is implemented now.
